graveler/committed/value_iterator.go

```diff
@@ -0,0 +1,45 @@
+package committed
+
+import "github.com/treeverse/lakefs/graveler"
+
+type valueIterator struct {
+	it Iterator
+}
+
+func (v *valueIterator) Next() bool {
+	for {
+		r := v.it.Next()
+		if !r {
+			return false
+		}
+		// skip ranges
+		_, rng := v.it.Value()
+		if rng == nil {
+			break
+		}
+	}
```

Comment:
This doesn't match the interface:
```go
	// Value returns a nil ValueRecord and a Range before starting a Range, or a Value and
	// that Range when inside a Range.
```
It should look like this (but I haven't tested it):
```suggestion
	for {
		r := v.it.Next()
		if !r {
			return false
		}
		// skip ranges
		val := v.it.Value()
		if val != nil {
			break
		}
	}
```

Reply:
Or easier for me to read perhaps:
```go
for v.it.Next() {
	if val := v.it.Value(); val != nil {
		return true
	}
}
return false
```

Reply:
Like this one!